### ISSUE TITLE
Support for 'git notes'

### ISSUE THREAD

**Issue by tohojo** [ctmarinas/stgit#20]
It would be useful if stgit could maintain `git notes` entries when manipulating commits. Since git attached notes to commit IDs, this would require stgit to port those whenever the ID changes (patch reorder, refresh, etc).

This is useful for keeping track of revision notes for upstream patch submission...

**Comment by ctmarinas** [comment]
I'd like this feature as well as I use git notes when modifying a patch. Below is an attempt at copying notes when the commit of a patch changes. If the approach sounds ok, I'll send a proper patch together with a test.

```
diff --git a/stgit/lib/git/repository.py b/stgit/lib/git/repository.py
index 4556c752c84d..bc68442827ac 100644
--- a/stgit/lib/git/repository.py
+++ b/stgit/lib/git/repository.py
@@ -428,3 +428,8 @@ class Repository(object):
     def repack(self):
         """Repack all objects into a single pack."""
         self.run(['git', 'repack', '-a', '-d', '-f']).run()
+
+    def copy_notes(self, old_sha1, new_sha1):
+        """Copy git notes from the old object to the new one."""
+        p = self.run(['git', 'notes', 'copy', old_sha1, new_sha1])
+        p.discard_exitcode().discard_stderr().discard_output()
diff --git a/stgit/lib/stack.py b/stgit/lib/stack.py
index efebe6b2d77f..d1f218c801e5 100644
--- a/stgit/lib/stack.py
+++ b/stgit/lib/stack.py
@@ -105,8 +105,14 @@ class Patch(object):
             pass
 
     def set_commit(self, commit, msg):
+        try:
+            old_sha1 = self.commit.sha1
+        except KeyError:
+            old_sha1 = None
         self._write_compat_files(commit, msg)
         self._stack.repository.refs.set(self._ref, commit, msg)
+        if old_sha1 and old_sha1 != commit.sha1:
+            self._stack.repository.copy_notes(old_sha1, commit.sha1)
 
     def delete(self):
         self._delete_compat_files()
```

**Comment by jpgrayson** [comment]
Closing.

I believe this was sufficiently addressed in commit 7031c39.
